demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -0,0 +1,625 @@
+                                          0 );
+        }
+        else
+        {
+            LogError( ( "Failed to add Range header to request headers: Error=%s.",
+                        HTTPClient_strerror( httpStatus ) ) );
+        }
+
+        if( httpStatus == HTTP_SUCCESS )
+        {
+            LogInfo( ( "Received HTTP response from %s%s...",
+                       pHost, pPath ) );
+            LogInfo( ( "Response Headers:\n%.*s",
+                       ( int32_t ) response.headersLen,
+                       response.pHeaders ) );
+            LogInfo( ( "Response Status:\n%u",
+                       response.statusCode ) );
+            LogInfo( ( "Response Body:\n%.*s\n",
+                       ( int32_t ) response.bodyLen,
+                       response.pBody ) );
+
+            /* We increment by the content length because the server may not have
+             * sent us the range we request. */
+            curByte += response.contentLength;
+
+            if( ( fileSize - curByte ) < numReqBytes )
+            {
+                numReqBytes = fileSize - curByte;
+            }
+        }
+        else
+        {
+            LogError( ( "Failed to send HTTP %s request to %s%s: Error=%s.",
+                        pMethod, pHost, pPath, HTTPClient_strerror( httpStatus ) ) );
+        }
+    }
+
+    if( httpStatus != HTTP_SUCCESS )
+    {
+        returnStatus = EXIT_FAILURE;
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of demo.
+ *
+ * This example resolves a domain, establishes a TCP connection, validates the
+ * server's certificate using the root CA certificate defined in the config header,
+ * then finally performs a TLS handshake with the HTTP server so that all communication
+ * is encrypted. After which, HTTP Client library API is used to download the
+ * S3 file by sending multiple GET requests, filling up the response buffer
+ * each time until all parts are downloaded. If any request fails, an error
+ * code is returned.
+ *
+ * @note This example is single-threaded and uses statically allocated memory.
+ *
+ */
+int main( int argc,
+          char ** argv )
+{
+    /* Return value of main. */
+    int32_t returnStatus = EXIT_SUCCESS;
+    /* HTTPS Client library return status. */
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+    /* The location of the path within string S3_PRESIGNED_URL. */
+    const char * pPath = NULL;
+    /* The length of the path within string S3_PRESIGNED_URL. */
+    size_t pathLen = 0;
+
+    /* The transport layer interface used by the HTTP Client library. */
+    TransportInterface_t transportInterface;
+    /* The network context for the transport layer interface. */
+    NetworkContext_t networkContext;
+
+    ( void ) argc;
+    ( void ) argv;
+
+    for( ; ; )
+    {
+        LogInfo( ( "HTTPS Client Synchronous S3 download demo using pre-signed URL:\n%s", S3_PRESIGNED_URL ) );
```

Comment:
The library name has changed since AFR
```suggestion
        LogInfo( ( "HTTP Client Synchronous S3 download demo using pre-signed URL:\n%s", S3_PRESIGNED_URL ) );
```